libcontainer/cgroups/utils.go

```diff
@@ -479,6 +479,26 @@ func RemovePaths(paths map[string]string) (err error) {
 	return fmt.Errorf("Failed to remove paths: %v", paths)
 }
 
+// RemovePathUnified aims to remove cgroup path recursively
+// Because there may be subcgroups in it.
+func RemovePathUnified(path string) error {
+	infos, err := ioutil.ReadDir(path)
+	if err == nil {
+		for _, info := range infos {
+			if info.IsDir() {
+				// We should remove subcgroups dir first
+				if err = RemovePathUnified(filepath.Join(path, info.Name())); err != nil && !os.IsNotExist(err) {
+					break
+				}
+			}
+		}
+		if err == nil || os.IsNotExist(err) {
+			return os.Remove(path)
```

Comment:
I think we should avoid returning `ENOENT` from here as well (similar to what `os.RemoveAll()` does.